src/test/java/sg/dex/starfish/integration/keeper/DirectPurchaseTest.java

```diff
@@ -0,0 +1,65 @@
+package sg.dex.starfish.integration.keeper;
+
+import org.junit.Test;
+
+import com.oceanprotocol.squid.exceptions.EthereumException;
+import org.web3j.crypto.CipherException;
+import java.io.IOException;
+
+import java.math.BigInteger;
+import com.oceanprotocol.squid.models.Account;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import sg.dex.starfish.impl.squid.SquidAccount;
+import sg.dex.starfish.impl.squid.SquidService;
+import sg.dex.starfish.keeper.DirectPurchaseAdapter;
+import com.oceanprotocol.squid.api.OceanAPI;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DirectPurchaseTest {
+    private DirectPurchaseAdapter directPurchaseAdapter;
+    private String reference;
+    private BigInteger tokenAmount;
+    String receiverAddress;
+    String senderAddress;
+
+    public DirectPurchaseTest() throws IOException, CipherException {
+        directPurchaseAdapter = new DirectPurchaseAdapter();
```

Comment:
If these are intended as constants for testing, just initialise them as static final constants above e.g.

```java
private static final String reference = "0x000000000000a46daef00000000000";
```

Reply:
Done